### PR TITLE
filebeat/input/unix: add input metrics support

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds metrics for unix socket packet count and total bytes, and histograms for time required to process TCP packets prior to acking from a publication and time between unix socket packet arrivals.

## Why is it important?

This allows us to help users configure their systems to match the requirements that they have.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: filebeat/input/unix/config.go [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-func defaultConfig() config {
-	return config{
-		Config: unix.Config{
-			Timeout:        time.Minute * 5,
-			MaxMessageSize: 20 * humanize.MiByte,
-			SocketType:     unix.StreamSocket,
-			LineDelimiter:  "\n",
-		},
-	}
-}
```
Comment (rdner) [inline filebeat/input/unix/config.go:41]: Why was it moved to `filebeat/input/unix/input.go`?

Comment (efd6) [reply filebeat/input/unix/config.go:41]: Code locality; having it in a separate file added no clarity.

File: filebeat/input/unix/input.go [hunk 2]
```diff
@@ -71,17 +93,27 @@ func (s *server) Test(_ input.TestContext) error {
 }
 
 func (s *server) Run(ctx input.Context, publisher stateless.Publisher) error {
-	log := ctx.Logger.Named("input.unix").With("path", s.config.Config.Path)
+	log := ctx.Logger.With("path", s.config.Config.Path)
```
Comment (belimawr) [inline filebeat/input/unix/input.go:96]: [Question]

Why removing the name of the logger? The name helps with filtering, specially when enabling log debug

File: filebeat/input/unix/config.go (hunk 1 above)
Comment (belimawr) [reply filebeat/input/unix/config.go:41]: While I generally agree with not keeping the config on a separate file, every (as far as I know) input on Filebeat uses a `config.go` file to hold it's configuration struct and the `defaultConfig` function.

To keep the project consistent I believe it's better to keep this file untouched.

Comment (efd6) [reply filebeat/input/unix/config.go:41]: Please take a look at udp and tcp.

File: filebeat/input/unix/input.go (hunk 2 above)
Comment (efd6) [reply filebeat/input/unix/input.go:96]: Please see https://github.com/elastic/beats/pull/33930#discussion_r1041709954

**Review by andrewkroh — APPROVED** [review]

File: filebeat/input/unix/config.go (hunk 1 above)
Comment (rdner) [reply filebeat/input/unix/config.go:41]: also, I would like to preserve the blame/changelog on the config-related file. Moving it to another file would discard all of that.

**Pull request merged**